instrument/types.go

```diff
@@ -38,4 +40,32 @@ type Options interface {
 
 	// GetMetricsScope returns the metricsScope
 	GetMetricsScope() tally.Scope
+
+	// GetGaugeInterval returns the time between updates of many gauges within the system
+	GetGaugeInterval() time.Duration
+}
+
+// MethodMetrics is a bundle of common metrics with a uniform naming scheme
+type MethodMetrics struct {
+	Error   tally.Counter
+	Success tally.Counter
+	Latency tally.Timer
+}
+
+// ReportSuccessOrFailure increments Error/Success counter dependant on the error
+func (m *MethodMetrics) ReportSuccessOrFailure(e error) {
```

Comment:
Move the implementation to a separate file? We usually only keep type defs in `types.go`